homeassistant/components/sensor/nsw_fuel_station.py

```diff
@@ -0,0 +1,120 @@
+"""
+Component to display the current fuel prices at a NSW fuel station.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/nsw_fuel_station/
+"""
+import datetime
+from typing import Optional
+import voluptuous as vol
+
+from homeassistant.components.light import PLATFORM_SCHEMA
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['nsw-fuel-api-client==1.0.7']
+
+CONF_STATION_ID = 'station_id'
+CONF_STATION_NAME = 'station_name'
+CONF_FUEL_TYPES = 'fuel_types'
+CONF_DEFAULT_FUEL_TYPES = ["E10", "U91", "E85", "P95", "P98", "DL",
+                           "PDL", "B20", "LPG", "CNG", "EV"]
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_STATION_ID): cv.positive_int,
+    vol.Required(CONF_STATION_NAME): cv.string,
```

Comment:
Must this really be `required`. `station_id` is used to retrieve the data incl. the station name. Best would be to let the users set their own name (`CONF_NAME`) and fall-back to the name which the request returns.

Reply:
Unfortunately the response from the API does not return the fuel station name:

https://api.onegov.nsw.gov.au/FuelCheckApp/v1/fuel/prices/station/2119

An alternative to this could be to just use the station id as the component name? What do you think is a better user experience?

Reply:
Hmm, I just checked one station from https://www.fuelcheck.nsw.gov.au and the name seems to be there.

```json
[{
  "ServiceStationID":1706,
  "Name":"BP Edgecliff",
  "Lat":-33.876974,
  "Long":151.231099,
[...]
```

Let the users have the option to name a sensor according their needs in the configuration will avoid that they have to use `customize:`. I personally prefer if the default name let me identify the sensor quickly, thus I would just use `Fuel Price` + fuel type. Station Id is not that useful, IMHO.

Reply:
Gotcha, I thought you were implying I obtained the station name by using the same request/endpoint. I will update the `setup_platform` to make an addition request during setup to find the name of the station using another endpoint. Definitely makes sense to make this easier to configure

> Let the users have the option to name a sensor according their needs in the configuration will avoid that they have to use customize:

I'll remove the need for a name in the config entirely



> thus I would just use Fuel Price + fuel type. Station Id is not that useful, IMHO.

Just double checking, I think you mean `Name` + fuel type?